### ISSUE TITLE
Best-move arrows showing different moves than engine tab

### ISSUE THREAD

**Issue by csaavedra** [veloce/lichobile#1826]
Sometimes the best-move arrows fall out of sync with the best moves in the engine tab. I think this happens often when the position is still being analyzed by the engine and often when the best lines are updated, the arrows will sync correctly again, but when they fall out of sync after reaching depth 24 and the engine stops analyzing, they stay in a different position than what the engine tab is showing.

![Screenshot_20210830-073414~2](https://user-images.githubusercontent.com/2523171/131285960-12560275-c481-4911-befb-a431531618d4.png)

If you are not looking at the engine tab, you might not even notice this.

**Comment by veloce** [comment]
Are you sure about this? Here in the screenshot I see 3 arrows corresponding to 3 lines.

**Comment by jas14** [comment]
I don't see the 21... Bf5 line shown, but the other two seem to be right

**Comment by csaavedra** [comment]
Yes, the best line `21... Bf5` is the one missing. IIRC that line was not among the best lines as of a lower depth, so that's probably why it wasn't showing. At depth 24 it is the best line, but the arrows are not up-to-date.

**Comment by csaavedra** [comment]
Here is another example where the best move `Be7` has no arrow. This at depth 24:

![Screenshot_20210910-073226~2](https://user-images.githubusercontent.com/2523171/132800424-de988c4d-62d8-40ff-aca9-7ed87767e838.png)

Then if you let the engine go deeper, after the evaluations are updated the arrows show up correctly:

![Screenshot_20210910-073843-345](https://user-images.githubusercontent.com/2523171/132800601-56725850-9f03-4ab9-ac89-c95b539140d9.png)

**Comment by veloce** [comment]
Thanks for all the details. I noticed that each time you have superposed arrows (here Nb4 for instance). This could lead to something.

**Comment by csaavedra** [comment]
Still happening:

![Screenshot_20220918-091915~2](https://user-images.githubusercontent.com/2523171/190888785-603da2a9-8451-4526-9f58-76a5eb9c2b95.png)

**Comment by phaebster** [comment]
is lichess dead? should I fork it? how can nobody address an issue in 2 years?

**Comment by jas14** [comment]
@phaebster https://lichess.org/blog/Y2Z47hEAAGUB7i7A/lichess-recruits-full-time-mobile-developer. You're welcome to create a fork and open a PR against this repository if you don't want to wait for the rewritten app.

**Comment by phaebster** [comment]
@jas14 thank you so much! I'd love to join lichess. I've played over 48k games here and I am software developer for over 20 years, also in eLearning domains.

**Comment by jas14** [comment]
@phaebster lichess is a free, open-source project/organization powered almost exclusively by volunteers, such as myself. To my knowledge, they are not hiring more developers; veloce was a very special case because he was the creator of the mobile app and has been its maintainer since its inception, and the organization grew large enough to require a full-time mobile developer. If you'd like to donate your time or money, please read https://lichess.org/help/contribute 🙂

**Comment by scha-ch** [comment]
@jas14 Thanks for your reply. Let's find out if there can be two special cases then. Reasons would be:
- 48k+ games at lichess
- 20y+ experience as a Software Engineer
- <3 chess

**grossmeister.ch** (not released but very soonish) will be a road-to-GM trainings app, I'm implementing if I'm not a 'special case' too. I've many experiences that are a perfect match, I'd work on it for free but somehow you gotta get a meal onto a table right.

I've planned and already implememted parts of:

-  Ankhorage: Web/Mobile Framework that generates mobile apps (React Native) and Webapp (NextJS for now, but supporting also VueUse and more as soon as I get some collaborators or a new pair of coding hands.) That means one codebase for mobile app and webapp.
- API Gateway: Providing chess API's e.g.
  - 'Open-Book', my just implemented opening book, my engine (Negamax AlphaBeta)
  - Neuronal network, so you can request a move (and configure what kind of move you want, e.g. randomly from book or depth 14 or best move within n seconds response time
  - Tools like PGN- FEN- conversion, validation, parsing, analyze a pgn and get a commented one...
- Engine teaches why a move is bad, it even speaks to you (speech synthesis model) if you want
- Bonus/Malus Chess App (if you play 'e4' wrong too often you won't be able to play that move anymore (but you'll get some lessons to accomplish first), so depending on how you use the app, the app could drastically adapt until you're worthy
- Opening Quiz: You have to draw and name the possible moves (and best move) of book move that have variations deeper down the tree, same for anchesters and ECO codes.
- Mini games to learn different topics in cool games, well Magnus App did so too, but there can be (many) more games.
- studies are checkable (at least) to see which games you've looked at and how many games still left to analyze
- Engine learns which kind of mistakes you do repeatedly and adapts lessons to make you improve more efficiently, suggests you books, courses, etc. (3rd party vendor program)
- Tactics by GM games (just the exclam moves for starters, then algorithm to auto-generate tactics, lichess already has this feature I assume)
- Bots playing styles can be configured in detail (they could be pawnstormer, piece centralizers, sacrifizers, knight lovers, daulphins (creative knight development bots), etc. etc.)
- 2FA (two factor authentication) will be a chess position ( !!! love that idea haha)
- there is way more, and a thousand ideas up next.

I have no problems implementing this on my own, but if lichess is doing the same thing, we could maybe at least have a quick meeting via zoom/meet/etc. to make sure we want 2 apps instead of 'the one and only'

Shouldn't be too difficult to find an investor if this is the problem, I mean lichess is that huge, should be possible to hire a whole team without problems.

I've also worked in the eLearning industry (Lead Frontend Engineer) so I'm probably most useful doing the thing I like most in my life with the people who implemented the app I like most (lichess)

**Comment by jas14** [comment]
@scha-ch it sounds like you are working on a lot of cool projects and are certainly passionate about chess! That being said, I think you misunderstand - I'm just a volunteer. I don't even maintain this repository, I just contribute for free sometimes. I'm certainly not among the people who would decide to hire you.

This being said, please also understand that Lichess is its community of almost entirely volunteers, most of whom have jobs elsewhere and contribute to Lichess through code, moderation, and in other ways in their free time.

You're welcome to come check out the [Lichess discord](https://discord.gg/lichess), but I'm afraid I can't help you re: employment.

**Comment by csaavedra** [comment]
> @jas14 (I've changed my username, phaebster here hehe, it's my chess github account) Thanks for your reply. Let's find out if there can be two special cases then. Reasons would be:

This is a bug report in an issue tracker, not the appropriate place for applying for a job and it certainly hurts your case if you can't realize this. Please keep the discussion on-topic.

**Comment by scha-ch** [comment]
@jas14 thank you for the discord link.
@csaavedra you're right, that was inappropriate.

Have a nice day everyone
